### PR TITLE
[Dashboard] Feature: Add Nebula Waitlist page

### PR DESCRIPTION
**Pull request by MananTank**
## Problem solved

Short description of the bug fixed or feature added

---

## PR-Codex overview
This PR introduces a new `Nebula` feature for team management, allowing users to join a waitlist. It includes new routes, components, and API integrations for handling waitlist functionality.

### Detailed summary
- Added `Nebula` route in `layout.tsx` files.
- Implemented `JoinNebulaWaitlistPage` component for waitlist UI.
- Created `joinTeamWaitlist` API function for server-side handling.
- Enhanced `getTeamNebulaWaitList` API function for fetching waitlist status.
- Updated `page.tsx` files to integrate waitlist functionality.
- Developed `JoinNebulaWaitlistPageUI` for user interaction with the waitlist.
- Added Storybook stories for `JoinNebulaWaitlistPageUI` component.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by MananTank** [comment]
* **#5406** <a href="https://app.graphite.dev/github/pr/thirdweb-dev/js/5406?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @MananTank and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/dashboard/src/app/team/[team_slug]/[project_slug]/nebula/components/nebula-waitlist-page-ui.client.tsx [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+function ShareButton() {
+  const [isCopied, setIsCopied] = useState(false);
+
+  return (
+    <ToolTipLabel label="Copy Page Link">
+      <Button
+        variant="outline"
+        className="gap-2"
+        onClick={() => {
+          navigator.clipboard.writeText("https://thirdweb.com/team/~/nebula");
```
[the remark this answers is not in the document]
Comment (MananTank) [reply apps/dashboard/src/app/team/[team_slug]/[project_slug]/nebula/components/nebula-waitlist-page-ui.client.tsx:55]: good bot

File: apps/dashboard/src/@/api/team.ts [hunk 2]
```diff
@@ -60,3 +57,31 @@ export async function getTeams() {
   }
   return [];
 }
+
+type TeamNebulWaitList = {
+  onWaitlist: boolean;
+  createdAt: null | string;
+};
+
+export async function getTeamNebuleWaitList(teamSlug: string) {
```
[the remark this answers is not in the document]
Comment (MananTank) [reply apps/dashboard/src/@/api/team.ts:66]: good bot

File: apps/dashboard/src/app/team/[team_slug]/[project_slug]/nebula/components/nebula-waitlist-page-ui.client.tsx [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+function ShareButton() {
+  const [isCopied, setIsCopied] = useState(false);
+
+  return (
+    <ToolTipLabel label="Copy Page Link">
+      <Button
+        variant="outline"
+        className="gap-2"
+        onClick={() => {
+          navigator.clipboard.writeText("https://thirdweb.com/team/~/~/nebula");
```
[the remark this answers is not in the document]
Comment (MananTank) [reply apps/dashboard/src/app/team/[team_slug]/[project_slug]/nebula/components/nebula-waitlist-page-ui.client.tsx:70]: its correct

**Review by jnsdls — APPROVED** [review]

**Pull request merged**